src/leveldb.nim

```diff
@@ -122,6 +122,22 @@ iterator iter*(self: LevelDb, seek: string = "", reverse: bool = false): (
     else:
       leveldb_iter_next(iterPtr)
 
+iterator iterPrefix*(self: LevelDb, prefix: string): (string, string) =
+  for (key, value) in iter(self, prefix, reverse = false):
+    if key.startsWith(prefix):
+      yield (key, value)
```

Comment:
Shouldn't there be `else: break`? Otherwise we will always iterate till the end of the database.

Reply:
Indeed!